apps/meteor/app/livechat/server/lib/QueueManager.ts

```diff
@@ -133,18 +133,42 @@ export class QueueManager {
 		return LivechatInquiryStatus.READY;
 	}
 
-	static async queueInquiry(inquiry: ILivechatInquiryRecord, room: IOmnichannelRoom, defaultAgent?: SelectedAgent | null) {
-		if (inquiry.status === 'ready') {
-			return RoutingManager.delegateInquiry(inquiry, defaultAgent, undefined, room);
-		}
-
+	static async notifyQueuedInquiry(inquiry: ILivechatInquiryRecord, room: IOmnichannelRoom, defaultAgent?: SelectedAgent | null) {
 		await callbacks.run('livechat.afterInquiryQueued', inquiry);
 
 		void callbacks.run('livechat.chatQueued', room);
 
 		await this.dispatchInquiryQueued(inquiry, room, defaultAgent);
 	}
 
+	static async queueInquiry(inquiry: ILivechatInquiryRecord, room: IOmnichannelRoom, defaultAgent?: SelectedAgent | null) {
+		if (!(await Omnichannel.isWithinMACLimit(room))) {
+			logger.error({ msg: 'MAC limit reached, not routing inquiry', inquiry });
+			// We'll queue these inquiries so when new license is applied, they just start rolling again
+			// Minimizing disruption
+			await saveQueueInquiry(inquiry);
+			await this.notifyQueuedInquiry(inquiry, room, defaultAgent);
+			return;
+		}
+
+		await callbacks.run('livechat.beforeRouteChat', inquiry, defaultAgent);
+		const dbInquiry = await LivechatInquiry.findOneById(inquiry._id);
+
+		if (!dbInquiry) {
+			throw new Error('inquiry-not-found');
+		}
+
+		if (dbInquiry.status === 'ready') {
+			logger.debug({ msg: 'Inquiry is ready. Delegating', inquiry, defaultAgent });
+			return RoutingManager.delegateInquiry(dbInquiry, defaultAgent, undefined, room);
+		}
+
+		if (dbInquiry.status === 'queued') {
+			logger.debug(`Inquiry with id ${inquiry._id} is queued. Notifying`);
+			await this.notifyQueuedInquiry(inquiry, room, defaultAgent);
+		}
```

Comment:
Work has been done to improve performance and to avoid finds and updates, isn't this undoing the job?
Just commenting because it doesn't seem like a mere fix, it seems like everything is wrong looking at the modifications.

Reply:
Yup, this is undoing some changes. Basically, when the perf changes were done, the call to:
```
await callbacks.run('livechat.beforeRouteChat', inquiry, defaultAgent);
```
was missing. 

This call is responsible for managing the `waiting queue` feature. It basically "queues" all incoming conversations, so the queueWorker can pick them up. Some MAC limitations were removed as well when the code started to call the new `queueInquiry`. The call for `notifyQueuedInquiry` it's because previously we used the `dispatchInquiryQueued`, which did a ton of things and it's now deprecated, now there's one doing similar stuff inside queuemanager which is good.

The bug, which i have not linked yet to the draft, is caused by this missing call, so even on EE + waiting queue enabled, no chats are being sent to the queue. Code does one attempt to assign, and then when it fails, it leaves the inquiry as "ready".

As well as with MAC limitations (Felipe is testing this) since the call to `withinLimits` was not done, a chat opened after an environment is MAC-restricted would be assigned, which shouldn't happen.

(i'm still checking if there's some way to reduce the calls, I think the callback returns an updated inquiry so that would be great, here i was just testing that with the changes everything else was working, which apparently is)

Reply:
These calls were still present on `requeueInquiry` (what a name), but this function is not called on the first chat assignment.

So, confirming that the feature comes back when the changes are reverted, confirmed they were introduced as part of the performance changes, I can start to improve from there and see a better solution.